server/util/fakeData.js

```diff
@@ -0,0 +1,46 @@
+const faker = require('faker');
+
+const dummyData = require('./dummydata.v2');
+
+
+const infoList = dummyData.map(rest => ({
+  id: rest.id,
+  name: rest.name,
+  seats: 20 + (Math.floor(50 * Math.random())),
+}));
+
+const generateReservations = (date) => {
+  const reservationList = [];
+
+  const dateObj = date || new Date();
+  const dateStr = dateObj.toISOString().slice(0, 10);
+
+  dummyData.forEach((rest) => {
+    const maxSeats = infoList.find(item => item.id === rest.id).seats;
+
+    // for 5pm to 9pm
+    for (let t = 17; t < 22; t += 1) {
+      // about 50% chance of fully booking a time
+      let seatsLeft = Math.min(maxSeats, Math.floor(maxSeats * 2 * Math.random()));
+
```

Comment:
why do you multiply maxSeats * 2 * Math.random()?

Reply:
All looks good otherwise, just trying to make sense of that line to understand how you handle the 'fullness' of each restaurant in your dummyData

Reply:
oh the maxSeats * 2 * random() is so that the random number chosen will be between 0 & twice the number of seats.  The Math.min will reduce it down to the maximum possible number of seats in the restaurant, so 50% of the time the reservation slot will be completely full & 50% of the time it will be a number less than maxSeats.